crates/aptos-dkg/src/range_proof.rs

```diff
@@ -133,11 +137,25 @@ pub fn commit<R>(pp: &PublicParameters, z: &[Scalar], rng: &mut R) -> (Commitmen
 where
     R: RngCore + rand::Rng + CryptoRng,
 {
-    let mut scalars = z.to_vec();
     let r = random_scalar(rng);
-    scalars.push(r);
-    let c = g1_multi_exp(&pp.lagr_g1, &scalars);
-    (Commitment(c), r)
+    let c = commit_with_randomness(pp, z, &r);
+    (c, r)
+}
+
+pub(crate) fn commit_with_randomness(
+    pp: &PublicParameters,
+    z: &[Scalar],
+    r: &Scalar,
+) -> Commitment {
+    let mut scalars = z.to_vec();
+    let mut bases: Vec<G1Projective> = pp.lagr_g1[..scalars.len()].to_vec();
+
+    scalars.push(*r);
+    let last_base = pp.lagr_g1.last().expect("pp.lagr_g1 must not be empty");
+    bases.push(*last_base);
+
+    let c = g1_multi_exp(&bases, &scalars);
```

Comment:
Why not call `g1_multi_exp(&pp.lagr_g1, &scalars);` as before? 

Not sure I follow why you are building the `bases` vectors up incrementally like this.